staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go

```diff
@@ -48,6 +49,10 @@ func WithAuthorization(handler http.Handler, requestContextMapper request.Reques
 		}
 		authorized, reason, err := a.Authorize(attributes)
 		if authorized == authorizer.DecisionAllow {
+			if len(reason) != 0 {
```

Comment:
would it make sense to annotate in all cases? right now, it's unclear whether a 403 came from authorization or a later admission rejection. Possibly something like:

```
audit.LogAnnotation(ae, "authorization.k8s.io/decision", "allow" or "forbid")
if len(reason) > 0 {
  audit.LogAnnotation(ae, "authorization.k8s.io/reason", reason)
}
if err != nil {
  audit.LogAnnotation(ae, "authorization.k8s.io/error", err.Error())
}
```

I don't know how detailed we want error/forbidden info to be, volume-wise

Reply:
`authorization.k8s.io/decision` is added.

`err` or `forbidden reason` will be recorded in `ResponseStatus` at `RequestResponse` level:
https://github.com/kubernetes/kubernetes/blob/e3d754a055be5c8327d3cfabe523187d8c52769d/staging/src/k8s.io/apiserver/pkg/apis/audit/types.go#L106

Will it be redundancy of information?

Reply:
> Will it be redundancy of information?

Not sure... I would anticipate most audit settings not including response content because it is very verbose

Reply:
Done.